dashboard/app/controllers/aichat_controller.rb

```diff
@@ -26,6 +30,37 @@ def chat_completion
     render(status: :ok, json: response_body)
   end
 
+  # params are newAichatEvent: AichatEvent, aichatContext: {currentLevelId: number; scriptId: number; channelId: string;}
+  # POST /aichat/log_aichat_event
+  def log_aichat_event
+    unless has_required_params?(AICHAT_ENDPOINTS[:log_aichat_event])
+      return render status: :bad_request, json: {}
+    end
+
+    context = params[:aichatContext]
+    event = params[:newAichatEvent]
+
+    project_id = nil
+    if context[:channelId]
+      _, project_id = storage_decrypt_channel_id(context[:channelId])
+    end
+
+    logged_event = AichatEvent.create(
```

Comment:
Do we need to handle a save failure? I think if we use .`create!` it should throw an exception if something goes wrong.

I also see in our codebase that we use `.new` followed by `.save!` but not sure if we have a preference

Reply:
Good call! I am also logging a jira ticket for myself to write dashboard unit tests for this and the fetch chat events controller action for a follow-up. https://codedotorg.atlassian.net/browse/LABS-952